spring-webflux/src/main/java/org/springframework/web/reactive/function/BodyInserters.java

```diff
@@ -72,19 +75,86 @@
 	}
 
 	/**
-	 * Inserter to write the given object.
-	 * <p>Alternatively, consider using the {@code syncBody(Object)} shortcuts on
+	 * Inserter to write the given object. The body can be one of the following:
+	 * <ul>
+	 * <li>Concrete value
+	 * <li>{@link Publisher} of value(s)
+	 * <li>Any other producer of value(s) that can be adapted to a
+	 * {@link Publisher} via {@link ReactiveAdapterRegistry}
+	 * </ul>
+	 * <p>Alternatively, consider using the {@code body(Object)} shortcuts on
 	 * {@link org.springframework.web.reactive.function.client.WebClient WebClient} and
 	 * {@link org.springframework.web.reactive.function.server.ServerResponse ServerResponse}.
-	 * @param body the body to write to the response
+	 * @param body the body to write
 	 * @param <T> the type of the body
 	 * @return the inserter to write a single object
 	 */
 	public static <T> BodyInserter<T, ReactiveHttpOutputMessage> fromObject(T body) {
+		Assert.notNull(body, "'body' must not be null");
+		ReactiveAdapter adapter = ReactiveAdapterRegistry.getSharedInstance().getAdapter(body.getClass());
```

Comment:
Using the shared instance means if someone tries to register additional adapters, they don't have a way to influence this.

Reply:
Can't we use the hints mechanism to resolve the registry?

Reply:
@rstoyanchev Could you please elaborate? I am not sure to understand.